cdap-support-bundle/bin/observability.py

```diff
@@ -0,0 +1,507 @@
+            "fileName": 'kubectl-exec-log-saver-logs-from-api',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/system/services/runtime/logs?start=0 -H "{auth_token}" ',
+            "fileName": 'kubectl-exec-runtime-logs-from-api',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/system/services/status -H "{auth_token}"',
+            "fileName": 'kubectl-exec-system-service-status',
+            "fileExt": 'txt',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_status
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/namespaces/system/apps/pipeline/services/studio/status -H "{auth_token}"',
+            "fileName": 'kubectl-exec-system-service-studio-status',
+            "fileExt": 'txt',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_status
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/namespaces/system/apps/dataprep/services/service/status -H "{auth_token}"',
+            "fileName": 'kubectl-exec-system-service-wrangler-status',
+            "fileExt": 'txt',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_status
+        }
+    ]
+
+    log_commands_processor(cdapCommandLists, directoryName, True, cdap_ns, pipeline_name, run_id)
+
+    update_progress(98)
+    compressed_file = shutil.make_archive(
+        base_name=directoryName,
+        format='gztar',
+        root_dir=parent_dir + directoryName
+    )
+
+    shutil.rmtree(directoryName)
+    update_progress(100)
+    print(f'Logs collection completed. Please find {directoryName}.tar.gz in current directory')
+    exit()
+
+
+def log_file_write(pathName, filename, fileWriteMode, contentToWrite, fileExt="log"):
+    f = open(f"{pathName}/{filename}.{fileExt}", fileWriteMode)
```

Comment:
use `with open()` to ensure that the file is always closed after we're done  - https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

Reply:
done